query-engine/core/src/query_graph_builder/write/nested/disconnect_nested.rs

```diff
@@ -165,14 +165,20 @@ fn handle_one_to_x(
     }
 
     // Depending on where the relation is inlined, we update the parent or the child nodes.
-    let (node_to_attach, model_to_update, extractor_model_id, null_record_id) =
+    let (node_to_attach, model_to_update, extractor_model_id, null_record_id, filter) =
         if parent_relation_field.is_inlined_on_enclosing_model() {
             // Inlined on parent
             let parent_model = parent_relation_field.model();
             let extractor_model_id = parent_model.primary_identifier();
             let null_record_id = SelectionResult::from(&parent_relation_field.linking_fields());
+            // If the relation is inlined on the parent and a filter is applied on the child, we need to join to apply this filter
+            let filter = if !filter.is_empty() {
+                parent_relation_field.to_one_related(filter)
+            } else {
+                filter
+            };
 
-            (parent_node, parent_model, extractor_model_id, null_record_id)
+            (parent_node, parent_model, extractor_model_id, null_record_id, filter)
```

Comment:
This is the fix for the nested disconnect bug. When the relation is inlined on the parent and a filter is applied on the child, we need to convert the filter to a relation filter so that a join is rendered. eg:

```prisma
model Parent {
  id Int @id

  childId Int? @unique
  childOpt Child? @relation(fields: [childId], references: [id])
}

model Child {
  id Int @id
  non_unique Int
  parentOpt?
}
```

```ts
await prisma.parent.update({
  where: { id: 1 },
  data: {
    childOpt: { disconnect: { non_unique: 1 } }
  }
})
```

This should roughly render:

```sql
UPDATE Parent SET childId = NULL
LEFT JOIN Child c ON (c.id = Parent.childId)
WHERE Child.non_unique = 1
```

Note: We do not use joins for relational filters. This some pseudo code.